### ISSUE TITLE
`clGetDeviceIds` doesnt recognize `CL_DEVICE_TYPE_ALL`

### ISSUE THREAD

**Issue by hughperkins** [jrprice/Oclgrind#98]
`clGetDeviceIds` doesnt recognize `CL_DEVICE_TYPE_ALL`

Therefore:
- if my library fetches only GPUs, it doesnt find oclgrind (since oclgrind claims to be a cpu)
- if my library fetches everything, ie `CL_DEVICE_TYPE_ALL`, it doesnt find oclgrind either :-)

**Comment by hughperkins** [comment]
(Note that using `CL_DEVICE_TYPE_GPU | CL_DEVICE_TYPE_ACCELERATOR | CL_DEVICE_TYPE_CPU` also fails)

**Comment by hughperkins** [comment]
So, the relevant code is here, in runtime.cpp: 
```
  if (device_type != CL_DEVICE_TYPE_CPU &&
      device_type != CL_DEVICE_TYPE_DEFAULT &&
      device_type != CL_DEVICE_TYPE_ALL)
  {
    ReturnError(NULL, CL_DEVICE_NOT_FOUND);
  }
```
I would argue that it should just pretend to be any device, and always return 1 device?

**Comment by hughperkins** [comment]
Note: this addresses also: https://github.com/jrprice/Oclgrind/issues/98

**Comment by jrprice** [comment]
Fixed by 60e429b.

**Comment by hughperkins** [comment]
Awesome.  Thanks!
